Response to issue 34

LTibbs opened the pull request:
I had some issues with the prokaryote trees, as described in Issue 34 (https://github.com/arendsee/phylostratr/issues/34). So, I made the following updates to address that:

1. Update `uniprot_strata()` to work with from=1 (starting from cellular organisms).
2. Create function `generate_prokaryote_tree` to build new prokaryotic tree from UniProt with customizable species numbers and weights.
3. Create function `use_custom_prokaryote_tree` to add the custom tree to the analysis, analogous to `use_recommended_prokaryotes`.

LTibbs committed:
Update uniprot-access.R to build prokaryote tree

LTibbs committed:
Update uniprot-access.R

LTibbs committed:
comment on tree_building.R

LTibbs committed:
Create functions to build new prokaryote trees

LTibbs committed:
Fix update_prokaryotes.R

LTibbs committed:
Add vignette using new prokaryote functions

arendsee commented:
@LTibbs Thanks for the PR. I appreciate the work you have put into this. I've been a bit under the rocks lately, but I'll look over your commits soon. I do have an old function for making prokaryotic trees: uniprot_sample_prokaryotes(remake=TRUE). Running that will create a new tree that is in harmony with they latest Uniprot annotation. This should avoid all the warnings you observed. I'll be back in touch soon.

arendsee commented:
@LTibbs I've updated the phylostratr functions `uniprot_sample_prokaryotes` (see c1d3deb9) and the `use_recommended_prokaryotes`. I'm happy to merge your contributions, but first see if these updated functions solve your problem. After pulling the latest from the main phylostratr branch, just update your code with `use_recommended_prokaryotes(version=2)`. I do the idea behind your `generate_prokaryote_tree` - it is more general than `uniprot_sample_prokaryotes` in that it allows weights and numbers to be set. Your update to `uniprot_strata` also looks reasonable, though I think the prokaryote outgroup will typically need special handling. So I'll keep this pull request open.

LTibbs commented:
@arendsee Thanks for pointing out the option to have `remake=T` in the `uniprot_sample_prokaryotes` function, and for your update to `use_recommended_prokaryotes`. That does solve my original problem of addressing warnings in the recommended prokaryotes. I know that some of the warnings in the earlier version were just due to not having a reference proteome, but others were more problematic (missing or "excluded" proteomes) so it's good to be able to update the tree easily as UniProt updates.

Still, I do think there is value in having the more general tree building function available for those who want to customize their prokaryote tree further, in both weights and size. I plan to use custom weights myself, so I think this branch and `generate_prokaryote_tree` could potentially be useful to others.

Depending on how things go, I could even combine the two approaches by adding an option for weights to the `uniprot_sample_prokaryotes` sampling?

LTibbs committed:
Update uniprot-access.R to enable exclusion of selected taxa

LTibbs commented:
@arendsee, I just finished code to add weights to the  `uniprot_sample_prokaryotes` function, but because it's based on your new branch, I think adding it now may cause conflicts. Could you merge my other changes, and then I can submit a new pull request on the most updated branch?

In addition, as I was working with phylostratr more, I found what was (to me) an unexpected result: even if I set weights for a given taxon to 0 or a negative number (i.e., intending to exclude it due to low quality proteome) it could still potentially be included in the final tree, depending on its position in the tree. So, I added a `drop.names` argument to `uniprot-access.R` in order to enable completely excluding certain taxa from the final tree.

arendsee commented:
Hi @LTibbs, your work here is awesome and super-appreciated; we'll get everything merged. But could we kick this down the road two weeks? I'm in serious crunch time at the moment working on a paper re-submission. Once that's done, I can allocate the time to re-familiarize myself with the `phylostratr` code and merge your work. Feel free to keep working on the code in the meantime, we'll get everything through eventually.

LTibbs committed:
Add weights to uniprot_sample_prokaryotes

LTibbs commented:
Of course! Hope your paper submission goes well. I really appreciate the work you did making this package, I'm just making these tweaks to help in my own use of the package and hope they can be useful for others!

LTibbs committed:
Add ability to drop taxa to uniprot_sample_prokaryotes

LTibbs committed:
add ability to drop specified taxa from generate_prokaryote_tree

LTibbs committed:
add drop.names argument

LTibbs committed:
fix typo uniprot-access.R

LTibbs committed:
Make ape library use explicit

LTibbs committed:
fix sample_taxids_weights

LTibbs committed:
Make ape library use explicit

LTibbs commented:
@arendsee I hope your paper submission went well! Have you had any more thoughts on merging the changes we discussed earlier in the year?

LTibbs commented:
@arendsee Just checking back in, would you have the time to collaborate on merging as discussed? Thanks!

arendsee committed:
Merge branch 'master' into master

arendsee commented:
@LTibbs It has been a long two weeks. Your updates all look reasonable, I'll go ahead and merge.

LTibbs commented:
Great, thanks! I just ran `roxygenise()` to incorporate the new functions and your `uniprot_sample_prokaryotes` into the package. I also added comments to the functions and fixed a bug in my `generate_prokaryote_tree`.
